### DISCUSSION TITLE
How to make output using secodary or multiple kernel using Egsnrc?

### DISCUSSION THREAD

**Discussion by kangdongsan** [nrc-cnrc/EGSnrc#1078] [Questions]
Hello, I was wondering how can I get the secondary or multiple kernel using EGSNRC.
Using EGS I can get the Total and Primary kernel using EGSNRC sample example file.

<img width="657" alt="image" src="https://github.com/nrc-cnrc/EGSnrc/assets/73975224/94f7ecd2-267b-48b8-be51-b257feea5c76">

Recently paper used just total or primary, but I want to get the more kernel data.
Any help would be greatly appreciated.

**Comment by rtownson** [comment]
The older algorithm in EDKnrc used by source 2 produces a file with all the components broken down, but it may suffer from numerical artifacts. The old algorithm encounters from possible round-off errors and unrealistic energy deposition of sub-threshold particles in the first region. Currently, source 0 only produces the total and primary EDK.

**Reply by kangdongsan** [reply]
Thank you for your reply. 
Is there any way at all to calculate secondary and multiples with the current version of egsnrc?

**Reply by rtownson** [reply]
Not that I know of - you just have the difference between the total and primary.

**Reply by kangdongsan** [reply]
Hi rtownson. 
Can I ask one more question?
I want to know more about ".egsdat" file. 
There are many values in".egsdat" file. but, I couldn't classified by meaning of values although I saw the manual.
----.egsdat file ----
<img width="1893" alt="image" src="https://github.com/nrc-cnrc/EGSnrc/assets/73975224/46fa0a0f-64ae-4174-9e57-c3ba26b2f2c3">
----manual----
![image](https://github.com/nrc-cnrc/EGSnrc/assets/73975224/2dbe3f9b-82df-4b53-930c-0be81e54d57b)


Thank you.

**Reply by rtownson** [reply]
The egsdat file is not meant to be used directly, which is why its format is not described. It's also different for each application. You could try to figure it out from reading the code, but I don't see why you would. All of the results from combining .egsdat files is output at the end of a simulation, so it handles reading the .egsdat files for you.

**Reply by kangdongsan** [reply]
While preprocessing the EGSlst file, I had a question. 
In ENERGY DEPOSITION KERNEL, I saw that each fraction is given by angle and radius, but I wanted to know how much energy is deposited in each voxel. Another command allows me to see the dose together, so I extracted the result. When I nomalized the fraction with dose, the result was different from the result of ENERGY DEPOSITION KERNEL.  Am I not understanding the output correctly? 
Any help would be greatly appreciated. I've attached the output below. 
[edknrc_template2.zip](https://github.com/nrc-cnrc/EGSnrc/files/14066901/edknrc_template2.zip)

**Reply by rtownson** [reply]
I'm not quite sure what you're comparing. In the energy deposition kernel, it lists the energy deposited in each voxel per source particle, not the fraction. Does that clarify things?

**Reply by kangdongsan** [reply]
Thank you as always. I understand exactly what I was confused about.
